python/upload.py

```diff
@@ -14,28 +15,44 @@ def create_new_branch(target_branch: str, source_branch: str = "main"):
     This function will create a new branch based on source_branch,
     named after target_branch. source_branch is default to main
     """
+    ref_target_branch = f"refs/heads/{target_branch}"
     src = repo.get_branch(source_branch)
-    repo.create_git_ref(ref=f"refs/heads/{target_branch}", sha=src.commit.sha)
+    repo.create_git_ref(ref=ref_target_branch, sha=src.commit.sha)
+    print(f"created branch: {ref_target_branch}")
 
 
-def upload_file(path_local: str, path_repo: str, target_branch: str, message: str = ""):
+def upload_file(path_local: str, target_branch: str):
     """
     this function will upload a given file to a given target_branch
     path_local: local file path relative to knownprojects_build
-    path_repo: relative file path within in the repo
     target_branch: the branch to commit files to
-    message: commit message that goes along with the file upload
     """
+    # relative file path within the repo
+    path_repo = _file.replace(basepath + "/", "")
+    # commit message that goes along with the file upload
+    message = f"🚀 {target_branch} -> {path_repo}..."
+
     with open(path_local, "rb") as f:
         content = f.read()
-    repo.create_file(path_repo, message, content, branch=target_branch)
+    print(f"uploading: {path_local} (local) to {target_branch}/{path_repo} (repo) ...")
+
+    try:
+        # try updating existing file
+        contents_existing = repo.get_contents(path_repo, ref=target_branch)
+        print(f"Updating exisitng file ...")
```

Comment:
The additional print statements make this much easier to read and will help us debug in the future if we have to

Reply:
Although I don't know how useful the sha print statement is, might be overkill

Reply:
true a little overkill. the sha is passed to `update_file()` though so it might be helpful when we have to debug